### PR TITLE
We need to change the SELinux label of the conmon process to s0

### PR DESCRIPTION
**Pull request by rhatdan**
If SELinux is enabled, we are leaking in pipes into the container
owned by conmon.  The container processes are not allowed to use
these pipes, if the calling process is fully ranged.  By changing
the level of the conmon process to s0, this allows container processes
to use the pipes.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Should fix https://github.com/projectatomic/libpod/issues/852

File: libpod/oci.go [hunk 1]
```diff
@@ -282,6 +284,20 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string) (er
...
+		// Set the label of the conmon process to be level :s0
+		// This will allow the container processes to talk to fifo-files
+		// passed into the container by conmon
+		plabel, err := selinux.CurrentLabel()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get current SELinux label")
+		}
+
+		c := selinux.NewContext(plabel)
+		c["level"] = "s0"
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply libpod/oci.go]: Well it is common on all platforms that support SELinux, except for ones that do not support level.  Which I think Gentoo might have a problem with.  Although without it you get less separation between containers.  I will look into making it optional for Gentoo.

**Comment by rhatdan** [comment]
@wking it should not handle SELinux installs that do not use "level" like gentoo.

File: libpod/oci.go [hunk 2]
```diff
@@ -282,6 +284,22 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, cgroupParent string) (er
...
+		// Set the label of the conmon process to be level :s0
+		// This will allow the container processes to talk to fifo-files
+		// passed into the container by conmon
+		plabel, err := selinux.CurrentLabel()
+		if err != nil {
+			return errors.Wrapf(err, "Failed to get current SELinux label")
+		}
+
+		c := selinux.NewContext(plabel)
+		if c["level"] != "s0" && c["level"] != "" {
```
Comment (TomSweeneyRedHat) [inline libpod/oci.go]: How would level be set to "" and wouldn't we want to set it to 's0' anyway?  This is probably a question exposing my selinux lack of knowledge.

**Comment by mheon** [comment]
LGTM once tests green

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@rh-atomic-bot r=mheon

File: libpod/oci.go (hunk 2 above)
Comment (wking) [reply libpod/oci.go]: > How would level be set to ""...

That's for [systems that don't support levels][1].

[1]: https://github.com/projectatomic/libpod/pull/854#discussion_r191910714

Comment (TomSweeneyRedHat) [reply libpod/oci.go]: Ah, that's makes sense.  Thanks for the ptr @wking

**Pull request closed without merging**